common/djangoapps/student/views.py

```diff
@@ -708,6 +711,11 @@ def dashboard(request):
         'course_programs': course_programs,
     }
 
+    referer = request.META.get('HTTP_REFERER')
+    if referer:
+        if referer.find("register") >= 0 or referer.find("oauth") > 0:
+            return redirect("/welcome-unactivated/")
+
```

Comment:
I don't think the `dashboard` view is the right place to have this login. This is only about user registration, not about rendering the dashboard.

Take a looks at [get_next_url_for_login_page](https://github.com/cognitiveclass/edx-platform/blob/master/common/djangoapps/student/helpers.py#L206), I think it makes much more sense to have the welcome redirect logic there.